include/jemalloc/internal/malloc_io.h

```diff
@@ -90,7 +90,22 @@ malloc_write_fd(int fd, const void *buf, size_t count) {
 }
 
 static inline ssize_t
-malloc_read_fd(int fd, void *buf, size_t count) {
+malloc_write_fd(int fd, const void *buf, size_t count) {
+	size_t bytes_written = 0;
+	while (bytes_written < count) {
+		ssize_t result = malloc_write_fd_syscall(fd,
+		    &((const byte_t *)buf)[bytes_written],
+		    count - bytes_written);
+		if (result == -1) {
```

Comment:
nit:maybe `if (result < 0) return result;` so it's not possible for bytes_written to underflow

Also maybe switch to `do {} while` since we at least call once.